### PR TITLE
Upgrade AGP to 7.2.2

### PR DESCRIPTION
**Pull request by vkryachko**

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.9-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.3.3-all.zip
```
Comment (thatfiredev) [inline gradle/wrapper/gradle-wrapper.properties:3]: Do we really need `all` or would `bin` suffice here?

File: encoders/protoc-gen-firebase-encoders/protoc-gen-firebase-encoders.gradle [hunk 2]
```diff
@@ -26,18 +26,25 @@ protobuf {
...
 
+kapt {
+    generateStubs = false
+}
+
 jar {
     manifest.attributes "Main-Class": "com.google.firebase.encoders.proto.codegen.MainKt"
 }
 
+kapt {
```
Comment (rlazo) [inline encoders/protoc-gen-firebase-encoders/protoc-gen-firebase-encoders.gradle:37]: could you merge both kapt sections?

**Review by rlazo — APPROVED** [review]

File: buildSrc/build.gradle.kts [hunk 3]
```diff
@@ -57,7 +56,9 @@ dependencies {
     implementation("org.apache.maven:maven-aether-provider:3.1.0")
 
     implementation("com.google.code.gson:gson:2.8.9")
-    implementation("com.android.tools.build:gradle:3.4.3")
+    implementation("org.jetbrains.dokka:dokka-android-gradle-plugin:0.9.17-g013-9b8280a")
```
Comment (thatfiredev) [inline buildSrc/build.gradle.kts:59]: Do we still need this now that dokka was removed?

Comment (vkryachko) [reply buildSrc/build.gradle.kts]: Great catch, I forgot to remove as part of the merge, removed

**Pull request merged**